pkg/autoupdate/findnew_test.go

```diff
@@ -396,6 +397,69 @@ func TestCheckExecutableTruncated(t *testing.T) {
 		"truncated binary")
 }
 
+func TestBuildTimestamp(t *testing.T) {
+	t.Parallel()
+
+	var tests = []struct {
+		buildTimestamp string
+		expectedNewest string
+		expectedOnDisk int
+	}{
+		{
+			buildTimestamp: "0",
+			expectedNewest: "5",
+			expectedOnDisk: 2,
+		},
+		{
+			buildTimestamp: "3",
+			expectedNewest: "5",
+			expectedOnDisk: 1, // remember, 4 is broken, so there should only be update 5 on disk
+		},
+		{
+			buildTimestamp: "5",
+			expectedOnDisk: 0,
+		},
+		{
+			buildTimestamp: "6",
+			expectedOnDisk: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run("buildTimestamp="+tt.buildTimestamp, func(t *testing.T) {
+			tmpDir, binaryName, cleanupFunc := setupTestDir(t, executableUpdates)
+			fmt.Println(tmpDir)
+			//defer cleanupFunc()
```

Comment:
do we want to run the cleanup func or remove this line?

Reply:
oops. That's development crud I left in. Nice catch.